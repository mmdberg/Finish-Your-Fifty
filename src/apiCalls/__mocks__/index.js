export const fetchUsers = jest.fn().mockImplementation(() => Promise.resolve(
  [{
    userName: 'Pizza',
    email: 'pizza@pizza',
    password: 'p'
  }]
));

export const addUser = jest.fn().mockImplementation(() => Promise.resolve(1));

export const fetchOneUser = jest.fn()
  .mockImplementationOnce(() => Promise.rejects())
  .mockImplementation(() => Promise.resolve({ 
    id: 7,
    userName: 'Pizza',
    email: 'pizza@pizza',
    password: 'p'
}));