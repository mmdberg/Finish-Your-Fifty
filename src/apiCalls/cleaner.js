export const raceCleaner = (races) => {
  const cleanRaces = races.map(race => {
    return {
      city: race.city,
      state: race.state,
      date: dateCleaner(race.start_date_local),
      raceName: race.name,
      id: race.id
    };
  });
  return cleanRaces;
};

export const dateCleaner = (date) => {
  const dateOnly = date.split('').splice(0, 10);
  const yearOnly = dateOnly.splice(0, 5).splice(0, 4);
  const cleanDate = [...date.split('').splice(5, 5), '-', ...yearOnly].join('');
  return cleanDate;
};