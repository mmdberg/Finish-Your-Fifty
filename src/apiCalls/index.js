import { apiKey } from '../private/apiKey';
import { raceCleaner } from './cleaner'

export const fetchRaces = async (state) => { 
  try {
    const response 
      = await fetch(`/v2/search/?query=running&category=event&state=${state}&api_key=${apiKey}`,
        {
          type: 'GET',
          credentials: 'omit'
        });
    const raceInfo = await response.json();
    return raceCleaner(raceInfo.results);
  } catch (error) {
    console.log('error:', error);
  }
};



export const fetchUsers = async () => {
  try {
    const response = await fetch('http://localhost:3000/api/v1/users/');
    const users = await response.json();
    return users;
  } catch (error) {
    console.log(error);
  }
};

export const fetchOneUser = async (id) => {
  try {
    const response = await fetch(`http://localhost:3000/api/v1/users/${id}`);
    const userInfo = await response.json();
    console.log(userInfo);
  } catch (error) {
    console.log(error);
  }
};

export const addUser = async (user) => {
  const response = await fetch('http://localhost:3000/api/v1/users', 
    {
      method: 'POST',
      body: JSON.stringify({
        userName: user.userName,
        email: user.email,
        password: user.password
      }),
      headers: {
        'Content-Type': 'application/json' 
      }
    });
  const userId = await response.json();
  return userId.id;
};

