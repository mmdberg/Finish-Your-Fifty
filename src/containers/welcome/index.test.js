import { shallow } from 'enzyme';
import { Welcome } from './index';
import React from 'react';
import * as api from '../../apiCalls';
// import { mockUser } from '../../mocks';

jest.mock('../../apiCalls');

describe('Welcome', () => {
  let wrapper;
  let mockMatch;
  let mockCaptureUser;

  beforeEach(() => {
    mockMatch = {path: '/welcome/signup'};
    mockCaptureUser = jest.fn();
    wrapper = shallow(<Welcome 
      match={mockMatch}
      captureUser={mockCaptureUser}/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should start with an empty state', () => {
    const expected = {
      userName: '',
      email: '',
      password: '',
      error: ''
    };
    expect(wrapper.state()).toEqual(expected);
  });

  it('should update state with user input', () => {
    const mockEvent = {target: {name: 'email', value: 'todd@todd'}};
    const expected = {
      userName: '',
      email: 'todd@todd',
      password: '',
      error: ''
    };

    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state()).toEqual(expected);
  });

  it('should call signUp if submit is clicked on signup page', () => {
    const mockEvent = { preventDefault: jest.fn() };
    wrapper.instance().signUp = jest.fn();

    wrapper.instance().handleSubmit(mockEvent);
    expect(wrapper.instance().signUp).toHaveBeenCalled();
  });

  it('should call logIn if submit is clicked on login page', () => {
    const mockEvent = { preventDefault: jest.fn() };
    const mockMatch= { path: '/welcome/login' };
    let wrapper = shallow(<Welcome match={mockMatch}/>);
    wrapper.instance().logIn = jest.fn();

    wrapper.instance().handleSubmit(mockEvent);
    expect(wrapper.instance().logIn).toHaveBeenCalled();
  });

  describe('signup', () => {
    it('should call fetchUsers on signup', async () => {
      await wrapper.instance().signUp();
      expect(api.fetchUsers).toHaveBeenCalled();
    });

    it('should clear inputs and set error state if email exists', async () => {
      const expected = {
        userName: '',
        email: '',
        password: '',
        error: 'This email already exists'
      };
      wrapper.setState({
        email: 'pizza@pizza'
      });
      await wrapper.instance().signUp();
      expect(wrapper.state()).toEqual(expected);
    });

    it('should call addUser with right params if new user', async () => {
      const expected = {
        userName: 'taco',
        email: 'taco@taco',
        password: 't',
        error: ''
      }
      wrapper.setState({
        userName: 'taco',
        email: 'taco@taco',
        password: 't'
      });

      await wrapper.instance().signUp();
      expect(api.addUser).toHaveBeenCalledWith(expected);
    });

    it('should call captureUser if new', async () => {
      const expected = {
        userName: 'taco',
        email: 'taco@taco',
        password: 't',
        id: 1
      }
      wrapper.setState({
        userName: 'taco',
        email: 'taco@taco',
        password: 't'
      });
      await wrapper.instance().signUp();
      expect(mockCaptureUser).toHaveBeenCalledWith(expected);
      //add right params
    });

    it('should reset state if new user', async () => {
      const expected = {
        userName: '',
        email: '',
        password: '',
        error: ''
      };
      wrapper.setState({
        userName: 'taco',
        email: 'taco@taco',
        password: 't'
      });
      await wrapper.instance().signUp();
      expect(wrapper.state()).toEqual(expected);
    });
  });

  describe('LogIn', () => {
    it('should set error state if email and password do not match', async () => {
      const expected = {
        userName: '',
        email: '',
        password: '',
        error: 'Email and password do not match. Please try again!'
      }
      await wrapper.instance().logIn({
        email: 'p@p',
        password: 'p'
      })
      expect(wrapper.state()).toEqual(expected)

    });

    it('should fetch user info with the right params', async () => {
      const mockCredentials = {
        email: 'pizza@pizza',
        password: 'p'
      }
      await wrapper.instance().logIn(mockCredentials);
      expect(api.fetchOneUser).toHaveBeenCalledWith(mockCredentials)
    });

    it('should call capture user', async () => {
      const mockCredentials = {
        email: 'pizza@pizza',
        password: 'p'
      }

      await wrapper.instance().logIn(mockCredentials);

      expect(mockCaptureUser).toHaveBeenCalledWith({    
        id: 7,
        userName: 'Pizza',
        email: 'pizza@pizza',
        password: 'p'
      })
    });
  });

});