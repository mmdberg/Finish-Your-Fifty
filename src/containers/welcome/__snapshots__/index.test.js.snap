// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome
    captureUser={[Function]}
    match={
        Object {
            "path": "/welcome/signup",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
>
        <h2>
                Please sign up to start tracking races
        </h2>
        <p
                className="welcome-text"
        >
                Have an account?
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="welcome-link"
                        to="/welcome/login"
                >
                        Log In
                </NavLink>
        </p>
        <input
                name="userName"
                onChange={[Function]}
                placeholder="Enter Name"
                type="text"
                value=""
        />
        <input
                name="email"
                onChange={[Function]}
                placeholder="Enter Email"
                type="email"
                value=""
        />
        <input
                name="password"
                onChange={[Function]}
                placeholder="Enter Password"
                type="password"
                value=""
        />
        <button
                type="submit"
        >
                Start Tracking
        </button>
        <p />
</form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Please sign up to start tracking races
</h2>,
          <p
            className="welcome-text"
>
            Have an account?
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="welcome-link"
                        to="/welcome/login"
            >
                        Log In
            </NavLink>
</p>,
          <input
            name="userName"
            onChange={[Function]}
            placeholder="Enter Name"
            type="text"
            value=""
/>,
          <input
            name="email"
            onChange={[Function]}
            placeholder="Enter Email"
            type="email"
            value=""
/>,
          <input
            name="password"
            onChange={[Function]}
            placeholder="Enter Password"
            type="password"
            value=""
/>,
          <button
            type="submit"
>
            Start Tracking
</button>,
          <p />,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Please sign up to start tracking races",
          },
          "ref": null,
          "rendered": "Please sign up to start tracking races",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Have an account?",
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="welcome-link"
                to="/welcome/login"
>
                Log In
</NavLink>,
            ],
            "className": "welcome-text",
          },
          "ref": null,
          "rendered": Array [
            "Have an account?",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Log In",
                "className": "welcome-link",
                "to": "/welcome/login",
              },
              "ref": null,
              "rendered": "Log In",
              "type": [Function],
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "userName",
            "onChange": [Function],
            "placeholder": "Enter Name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "onChange": [Function],
            "placeholder": "Enter Email",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "password",
            "onChange": [Function],
            "placeholder": "Enter Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Start Tracking",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Start Tracking",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
>
          <h2>
                    Please sign up to start tracking races
          </h2>
          <p
                    className="welcome-text"
          >
                    Have an account?
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="welcome-link"
                              to="/welcome/login"
                    >
                              Log In
                    </NavLink>
          </p>
          <input
                    name="userName"
                    onChange={[Function]}
                    placeholder="Enter Name"
                    type="text"
                    value=""
          />
          <input
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter Email"
                    type="email"
                    value=""
          />
          <input
                    name="password"
                    onChange={[Function]}
                    placeholder="Enter Password"
                    type="password"
                    value=""
          />
          <button
                    type="submit"
          >
                    Start Tracking
          </button>
          <p />
</form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Please sign up to start tracking races
</h2>,
            <p
              className="welcome-text"
>
              Have an account?
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="welcome-link"
                            to="/welcome/login"
              >
                            Log In
              </NavLink>
</p>,
            <input
              name="userName"
              onChange={[Function]}
              placeholder="Enter Name"
              type="text"
              value=""
/>,
            <input
              name="email"
              onChange={[Function]}
              placeholder="Enter Email"
              type="email"
              value=""
/>,
            <input
              name="password"
              onChange={[Function]}
              placeholder="Enter Password"
              type="password"
              value=""
/>,
            <button
              type="submit"
>
              Start Tracking
</button>,
            <p />,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Please sign up to start tracking races",
            },
            "ref": null,
            "rendered": "Please sign up to start tracking races",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Have an account?",
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="welcome-link"
                  to="/welcome/login"
>
                  Log In
</NavLink>,
              ],
              "className": "welcome-text",
            },
            "ref": null,
            "rendered": Array [
              "Have an account?",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Log In",
                  "className": "welcome-link",
                  "to": "/welcome/login",
                },
                "ref": null,
                "rendered": "Log In",
                "type": [Function],
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "userName",
              "onChange": [Function],
              "placeholder": "Enter Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "Enter Email",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "Enter Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Start Tracking",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Start Tracking",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
